core/camera/src/main/java/com/google/jetpackcamera/core/camera/CameraXCameraUseCase.kt

```diff
@@ -560,12 +565,24 @@ constructor(
                 .start(callbackExecutor) { onVideoRecordEvent ->
                     run {
                         Log.d(TAG, onVideoRecordEvent.toString())
+                        val isTorchRequired = camera.cameraInfo.hasFlashUnit() &&
+                            currentSettings.value?.flashMode == FlashMode.ON
                         when (onVideoRecordEvent) {
+                            is VideoRecordEvent.Start -> {
+                                if (isTorchRequired) {
+                                    camera.cameraControl.enableTorch(true)
+                                }
+                            }
                             is VideoRecordEvent.Finalize -> {
+                                if (isTorchRequired) {
+                                    camera.cameraControl.enableTorch(false)
+                                }
                                 when (onVideoRecordEvent.error) {
                                     ERROR_NONE ->
                                         onVideoRecord(
-                                            CameraUseCase.OnVideoRecordEvent.OnVideoRecorded
+                                            CameraUseCase.OnVideoRecordEvent.OnVideoRecorded(
+                                                onVideoRecordEvent.outputResults.outputUri
+                                            )
```

Comment:
We've been meaning to put this entire function into its own coroutine within the `runCamera` block so we can get rid of the shared instance variables for `videoCaptureUseCase` and `recording`, similar to what we did [here](https://github.com/google/jetpack-camera-app/pull/202/files#diff-8f5d934183d510bc928fe366b2c1bf6924dcbdc8afae80131a05ea9d469195eeR133)

It's not absolutely necessary for this PR, but if you would like it could make this a bit cleaner.

We would use a `Channel<VideoCaptureControlEvent>` where `VideoCaptureControlEvent` is a private sealed interface that has start and stop events. We could either pass the callback from the arguments of this function into the start event.

Reply:
Update the code as needed. However, I discovered an issue in the existing code:
```
launch {
    focusMeteringEvents.consumeAsFlow().collect {
        ...
    }
}
```
This code has a bug. When you log the associated camera lens facing, you'll notice that after switching the camera and tapping to focus, the focus events may sent to the wrong camera for each tap.
```
Log.d(TAG, "Starting focus and metering " + camera.cameraInfo.lensFacing)
```
The root cause seems to be that the code launches a new coroutine and listener for the channel every time the camera lens is switched (or any setting is changed). This results in multiple coroutines running concurrently, all doing the same thing but with different local variables. While the "tap to focus" issue isn't immediately apparent, it becomes easily noticeable when implementing video controls, especially after switching cameras.

The new commit attempts to address the issue by canceling the previous collection of the video control channel, however, this might not be as elegant as you expected.

Reply:
That's odd. Switching cameras should cancel the previous camera coroutine (and all child coroutines due to structured concurrency). If that's not happening then we should figure out why. There shouldn't be a need to close the channel and create a new one. That actually defeats the purpose of using a channel for synchronization since it's just a mutable instance variable at that point which could technically be called from different threads.